### PR TITLE
Added code for finding the maximum elememt in a Bitonic Array

### PR DESCRIPTION
**Pull request by kotwani2883**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added an optimized code for finding the maximum elememt in a Bitonic Array.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2395

### CODE REVIEW

File: C-Plus-Plus/other/4Sum.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/*
+   Time Complexity: O(n)
+   Auxilary Space Complexity: O(n)
+
+   Input:  n=6
+           sum=91
+           a[n]={10 ,20 ,30 ,40 ,1 ,2}
+   Output: 20, 30, 40, 1
+*/
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp]: Remove the file

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      return findMax(arr, low, mid-1); 
+   else /* when arr[mid] is greater than arr[mid-1] and smaller than arr[mid+1] */
+      return findMax(arr, mid + 1, high);   
+}
+
+
+//Drivers' Code
+int main()
+{
+	int n;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: Write good naming convention

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	cin>>n;
+	int arr[n];
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: Write good naming convention

Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	for(int i=0;i<n;i++)
+	  cin>>arr[i];
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp:58]: Display a message in print form as what are you taking input from the user

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	cout<<"The maximum element is: "<<findMax(arr,0,n-1);
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: add new line at the end of the output

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+Corner case (No decreasing part)
+Input: arr[] = {10, 20, 30, 40, 50}
+Output: 50
+
+Testcase '3':
+Corner case (No increasing part)
+Input: arr[] = {120, 100, 80, 20, 0}
+Output: 120
+*/
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp:84]: Write Time & Space Complexity at the end of the code

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+Find the maximum element in Bitonic array
+==========================================
+A bitonic array is an array of integers which is initially
+increasing and then decreasing.
+An efficient solution to find the maximum element 
+in bitonic array is by using binary search.
+Given an array of integers which is initially increasing
+and then decreasing, find the maximum value in the array.
+
+Time Complexity: O(Logn)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: Shift it to the end & also include space complexity

File: C-Plus-Plus/README.md [hunk 8]
```diff
@@ -280,8 +280,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md]: Remove these lines

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Do the suggested changes

File: C-Plus-Plus/README.md [hunk 9]
```diff
@@ -371,7 +370,4 @@ _add list here_
 - [KnapSack Fractional using Greedy Approach](other/knapsackfraction.cpp)
 - [Longest Subarray with equal number of 0 and 1](other/Longest_subarray_with_equal_0_1.cpp)
 - [Set Clear and Check bits](other/check_set_clear_bit.cpp)
-- [Iterative approach to print all subsets](other/subsets_iterative.cpp)
-- [Count of Set Bits in an Integer using Lookup Technique](other/count_of_set_bit_using_lookup_tech.cpp)
-- [Concatenate given strings](other/n_string_concatenate.cpp)
-- [Trapping Rainwater](other/trappingRainwater.cpp)
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md]: unnecessary deletion of the previous content,
Add it

File: C-Plus-Plus/README.md [hunk 10]
```diff
@@ -284,8 +284,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (siddharth25pandey) [inline C-Plus-Plus/README.md]: Same content twice in a line

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Minor changes is required

**Comment by kotwani2883** [comment]
Kindly Check. I have made the changes.
Thanks!!

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    Testcase '2':
+    Corner case (No decreasing part)
+    Input: arr[] = {10, 20, 30, 40, 50}
+    Output: 50
+
+    Testcase '3':
+    Corner case (No increasing part)
+    Input: arr[] = {120, 100, 80, 20, 0}
+    Output: 120
+*/
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/Max_in_BitonicArray.cpp]: Please change according to program.
There is no print message in Sample Input & Output

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kotwani2883 Minor changes required

File: C-Plus-Plus/other/Max_in_BitonicArray.cpp (hunk 11 above)
Comment (kotwani2883) [reply C-Plus-Plus/other/Max_in_BitonicArray.cpp]: I am unable to resolve the conflicts on my machine .What to do in this case.

File: C-Plus-Plus/other/4Sum.cpp [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+	for (int i = 0; i < n - 1; i++) {
+		for (int j = i + 1; j < n; j++)
+			ump[arr[i] + arr[j]] = {i, j};
+	}
+
+	/* Traverse through all pairs and search for X - (current pair sum) */
+	for (int i = 0; i < n - 1; i++) {
+		for (int j = i + 1; j < n; j++) {
+			int X = arr[i] + arr[j];
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/other/4Sum.cpp]: Remove the redundant file

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Review by siddharth25pandey — APPROVED** [review]
@kotwani2883 LGTM 💯 able to merge

**Pull request merged**